src/openms/source/CHEMISTRY/NASequence.cpp

```diff
@@ -267,12 +267,14 @@ namespace OpenMS
 
   double NASequence::getMonoWeight(NASFragmentType type, Int charge) const
   {
-    return getFormula(type, charge).getMonoWeight();
+    //getFormula adds Hydrogens, not protons, so we need to subtract the electrons
+    return getFormula(type, charge).getMonoWeight() - charge * Constants::ELECTRON_MASS_U;
   }
 
   double NASequence::getAverageWeight(NASFragmentType type, Int charge) const
   {
-    return getFormula(type, charge).getAverageWeight();
+    //getFormula adds Hydrogens, not protons, so we need to subtract the electrons
+    return getFormula(type, charge).getAverageWeight() - charge * Constants::ELECTRON_MASS_U;
```

Comment:
What if charge is negative? Is still the correct mass calculated?

Reply:
Yes, in the negative charge case getFormula returns the formula of the uncharged species minus the hydrogens, so we need to add back the electron mass difference between the hydrogen and the proton, this is accomplished here by subtracting a negative number of ELECTRON_MASS_U

Reply:
Can you clarify this in the comment. Sounds like it adds always